source/adios2/transform/compress/CompressBZip2.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * CompressBZip2.h : wrapper to BZip2 compression library
+ *
+ *  Created on: Jul 24, 2017
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+
+#ifndef ADIOS2_TRANSFORM_COMPRESSION_COMPRESSBZIP2_H_
+#define ADIOS2_TRANSFORM_COMPRESSION_COMPRESSBZIP2_H_
+
+#include <bzlib.h>
+
+#include "adios2/core/Transform.h"
+
+namespace adios2
+{
+namespace transform
+{
+
+class CompressBZip2 : public Transform
+{
+
+public:
+    /**
+     * Unique constructor
+     * @param debugMode
+     */
+    CompressBZip2(const bool debugMode);
+
+    ~CompressBZip2() = default;
```

Comment:
Parent classes can never be final, hence the destructor of Transport is virtual, but marking anything virtual in a final class like CompressBZip2 has no effect. https://isocpp.org/wiki/faq/virtual-functions#virtual-dtors   PbZip2 already exists http://compression.ca/pbzip2/ and is orthogonal to this class as it's a library on its own. Now, the final keyword reflects today's intention (ADIOS.cpp will create a CompressBZip2), making this class a Base class will be as simple as removing the final keyword and adding virtual to the destructor and intended functions. I am not against your idea, but we should do that when we cross that bridge.